### PR TITLE
Issue with wrong IBOutlet in ProductsViewController.swift solved.

### PR DESCRIPTION
**Pull request by CosmicYogi**
### What this does

The Sample application was crashing when were selecting the collection and when it was going to `ProductsViewController` because the IBOutlet for CollectionView was not configured properly.

The CollectionView is added in the ViewController in which it should be i.e.., `ProductsViewController` and IBoutlet connections are done properly.

So now the sample application is working fine without any crash.

### CODE REVIEW

File: Sample Apps/Storefront/Storefront/Base.lproj/Main.storyboard [hunk 1]
```diff
@@ -139,11 +138,12 @@
                         <rect key="frame" x="0.0" y="0.0" width="375" height="667"/>
                         <autoresizingMask key="autoresizingMask" widthSizable="YES" heightSizable="YES"/>
                         <subviews>
-                            <collectionView clipsSubviews="YES" multipleTouchEnabled="YES" contentMode="scaleToFill" alwaysBounceVertical="YES" delaysContentTouches="NO" dataMode="prototypes" translatesAutoresizingMaskIntoConstraints="NO" id="l3f-iD-z0W" customClass="StorefrontCollectionView" customModule="Storefront" customModuleProvider="target">
+                            <collectionView clipsSubviews="YES" multipleTouchEnabled="YES" contentMode="scaleToFill" misplaced="YES" alwaysBounceVertical="YES" delaysContentTouches="NO" dataMode="prototypes" translatesAutoresizingMaskIntoConstraints="NO" id="w1D-iY-CwD" customClass="StorefrontCollectionView" customModule="Storefront" customModuleProvider="target">
```
Comment (dbart01) [inline Sample Apps/Storefront/Storefront/Base.lproj/Main.storyboard]: Please don't commit `misplaced="YES"` views.

File: Sample Apps/Storefront/Storefront/Base.lproj/Main.storyboard [hunk 2]
```diff
@@ -154,19 +154,13 @@
...
                                 </connections>
                             </collectionView>
                         </subviews>
                         <color key="backgroundColor" white="0.97193080357142858" alpha="1" colorSpace="calibratedWhite"/>
-                        <constraints>
-                            <constraint firstAttribute="trailing" secondItem="l3f-iD-z0W" secondAttribute="trailing" id="971-jO-5TT"/>
-                            <constraint firstItem="oTA-Tk-tHe" firstAttribute="top" secondItem="l3f-iD-z0W" secondAttribute="bottom" id="9Pu-yp-tHC"/>
-                            <constraint firstItem="l3f-iD-z0W" firstAttribute="leading" secondItem="xKM-7h-oPY" secondAttribute="leading" id="aFP-HO-uyh"/>
-                            <constraint firstItem="l3f-iD-z0W" firstAttribute="top" secondItem="xKM-7h-oPY" secondAttribute="top" id="t6s-Aa-fox"/>
-                        </constraints>
```
Comment (dbart01) [inline Sample Apps/Storefront/Storefront/Base.lproj/Main.storyboard]: Why are these auto-layout constraints removed in favour of `autoresizingMask`?

**Comment by CosmicYogi** [comment]
Yeah,
It fixes the connections in storyboard in the Sample Application.
If you check out the `ProductsViewController` there is an Outlet `@IBOutlet weak var collectionView: StorefrontCollectionView!`
It was not connected properly with the Main storyboard.
If you run the sample app you will face a crash when you click on collection.
It is because in `configureCollectionView()` the delegate is set for this collection view by this line of code `self.collectionView.paginationDelegate = self`
And because it is not connected with storyboard properly and it is used it crashes.

So what I did was - corrected those wrong connections.

**Comment by dbart01** [comment]
screen shot 2018-04-18 at 9 47 20 am

I can't reproduce your crash. The `collectionView` is connected and loading correctly, as seen in the screenshot above.

**Comment by CosmicYogi** [comment]
Hey, I tried working with the repo on a different laptop.
It was actually happening because of an issue in Xcode so I reported that bug to Apple.
Then I also tried few other tests after watching your comment they were working fine on Xcode on another machine.
So, I think we should close this.

**Comment by dbart01** [comment]
Okay, I'll leave up to you to close. Thanks anyway for your PR and initiative!

**Pull request closed without merging**